src/server/bg_services/lifecycle.js

```diff
@@ -16,80 +16,67 @@ var LIFECYCLE = {
     schedule_min: 1 //run every 5 minutes
 };
 
-function background_worker() {
+async function background_worker() {
     const system = system_store.data.systems[0];
-    if (!system) return P.resolve();
-    return P.fcall(function() {
-            dbg.log0('LIFECYCLE READ BUCKETS configuration:', 'BEGIN');
-            return system_store.refresh()
-                .then(function() {
-                    _.each(system_store.data.buckets, function(bucket, i) {
-                        if (!bucket.lifecycle_configuration_rules || bucket.deleting) {
-                            return;
-                        }
-                        P.all(_.map(bucket.lifecycle_configuration_rules, function(lifecycle_rule, j) {
-                            dbg.log0('LIFECYCLE HANDLING BUCKET:', bucket.name, 'BEGIN');
-                            var now = Date.now();
-                            //If refresh time
-                            if (!lifecycle_rule.last_sync) {
-                                dbg.log0('LIFECYCLE HANDLING bucket:', bucket.name, 'rule id', lifecycle_rule.id,
-                                    'status:', lifecycle_rule.status, ' setting yesterday time');
-                                lifecycle_rule.last_sync = now - (1000 * 60 * 60 * 24); //set yesterday as last sync
-                            }
-                            dbg.log0('LIFECYCLE HANDLING bucket:', bucket.name, 'rule id(', j, ')', lifecycle_rule.id,
-                                'status:', lifecycle_rule.status, 'last_sync',
-                                Math.floor((now - lifecycle_rule.last_sync) / 1000 / 60), 'min ago.');
+    if (!system) return;
+    try {
+        dbg.log0('LIFECYCLE READ BUCKETS configuration: BEGIN');
+        await system_store.refresh();
+        for (const bucket of system_store.data.buckets) {
+            if (!bucket.lifecycle_configuration_rules || bucket.deleting) return;
+
+            P.all(_.map(bucket.lifecycle_configuration_rules, async function(lifecycle_rule, j) {
```

Comment:
check if this has to be in the background or we should wait for it.

also maybe change to arrow function 
```suggestion
            P.all(_.map(bucket.lifecycle_configuration_rules, async (lifecycle_rule, j) => {
```